### ISSUE TITLE
Getting smart object properties 

### ISSUE THREAD

**Issue by milqnomadu** [psd-tools/psd-tools#422]
I know, athough incredibly surprising, that it's not possible to write inside SOs.

However would it be possible instead to retrieve the properties applied to the smart object (perspective change, warp, blend options, opacity, position, dimensions) and use those in a python script that would take those properties and create its own smart object placeholder mask, which in this case can be used to replaced overlaying an image applying the SO property?

**Comment by milqnomadu** [comment]
?

**Comment by milqnomadu** [comment]
?

**Comment by spector700** [comment]
@milqnomadu Did you get this working?

**Comment by dadwjiodjwad** [comment]
@spector700 can’t get warp properties in any way unfortunately

**Comment by spector700** [comment]
This is a smart layer and I opened with PIL. I get the transform_box chords from the smart object, and if I draw them it is incorrect. Why is this?

```python
    base_image: Image = smart_layer.smart_object.topil()
    points = [
        (649.8286674132139, 2178.74013434089),
        (3468.5239616613417, 1068.2773109243699),
        (4847.65092073659, 1584.6679765431352),
        (2068.7717049038592, 3137.0252100840335),
    ]
    draw = Draw(base_image)
    draw.line([points[0], points[1], points[2], points[3], points[0]], "blue", 10)
```
![tmp9fjjbyxn](https://github.com/user-attachments/assets/c3cb47fc-d641-4483-9742-29a0eabd2bcd)

**Comment by milqnomadu** [comment]
@spector700 have you managed to solve this?

**Comment by spector700** [comment]
Yeah, I did get it aligned. But I switched to using the [Photopea API](https://www.photopea.com/api/live), which is in JavaScript. I needed to add the image into the smart object and not just replace over it, which didn't seem possible with psd-tools.

**Comment by milqnomadu** [comment]
@spector700 would it be possible for you to share what you used with Photopea? I wanted to use it swell but I didn't manage to make it work. If so please we can get in touch via email at milqnomadu@gmail.com. Thankss!
